python/ray/metrics_agent.py

```diff
@@ -181,3 +187,65 @@ def _register_if_needed(self, metric_point: MetricPoint):
                 metric_name].view.measure._description = metric_description
             self._registry[metric_name].view.measure._unit = metric_units
             self._missing_information = False
+
+
+class PrometheusServiceDiscoveryFileWriter:
+    def __init__(self, temp_dir):
+        self.temp_dir = temp_dir
+
+    def write(self):
+        # Write a file based on https://prometheus.io/docs/guides/file-sd/
+        nodes = ray.nodes()
+        metrics_export_addresses = [
+            "{}:{}".format(node["NodeManagerAddress"],
+                           node["MetricsExportPort"]) for node in nodes
+        ]
+        data = [{
+            "labels": {
+                "job": "ray"
+            },
+            "targets": [
+                metrics_export_address
+                for metrics_export_address in metrics_export_addresses
+            ]
+        }]
+        with open(self.get_target_file_name(), "w") as json_file:
+            json.dump(data, json_file)
+
+    def get_target_file_name(self):
+        return os.path.join(
+            self.temp_dir, ray.ray_constants.PROMETHEUS_SERVICE_DISCOVERY_FILE)
```

Comment:
This doesn't need to be in its own class. The abstraction and it's interaction with exception handling in the while True block in the thread is leaky.

Reply:
Right. That's a good point. Let me merge two classes.